### PR TITLE
implement Fmt<Display> for all the integer primitives

### PR DESCRIPTION
**Pull request by japaric**
using numtoa

### CODE REVIEW

File: src/int_impls.rs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                fn size_hint(&self, _strategy: &Display) -> usize {
+                    $N
+                }
+            }
+        )+
+    }
+}
+
+// NOTE formatting negative integers requires one more byte than necessary.
+// See mmstick/numtoa#8.
```
Comment (Kixunil) [inline src/int_impls.rs]: Increasing N by 1 fixes `numtoa` bug but creates incorrect `size_hint`.

File: src/int_impls.rs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                fn fmt<W>(
+                    &self,
+                    writer: &mut W,
+                    _strategy: &Display,
+                ) -> Result<(), W::Error>
+                where
+                    W: Write,
+                {
+                    unsafe {
+                        let mut buffer: [u8; $N] = mem::uninitialized();
```
Comment (Kixunil) [inline src/int_impls.rs]: I suggest using `$N + 1` here until the `numtoa` bug is fixed.

File: src/int_impls.rs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+// See mmstick/numtoa#8.
+impls! {
+    (i8, 5),
+    (i16, 7),
+    (i32, 12),
+    (i64, 21),
+    (u8, 3),
+    (u16, 5),
+    (u32, 10),
+    (u64, 20),
```
Comment (Kixunil) [inline src/int_impls.rs]: After `$N + 1` is used for allocating buffer, signed integers should use correct `size_hint`

**Review by Kixunil — CHANGES_REQUESTED** [review]
Please use correct size_hint and adjust only allocation for `numtoa` bug. Alternatively, if fixed version of `numtoa` is released, just change `$N` for signed integers.

**Comment by japaric** [comment]
@Kixunil fixed size_hint

**Comment by Kixunil** [comment]
Nice, thank you!

**Pull request merged**